kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NMDeviceStateTest.java

```diff
@@ -0,0 +1,181 @@
+    	thenIsConnectShouldReturn(false);
+    }
+    @Test
+    public void shouldReturnConnectionFalseUnavailable() throws InterruptedException, KuraException {
+    	whenStateIsSetTo(NMDeviceState.NM_DEVICE_STATE_UNAVAILABLE);
+    	thenIsConnectShouldReturn(false);
+    }
+    
+    @Test
+    public void shouldReturnConnectionFalseDisconnected() throws InterruptedException, KuraException {
+    	whenStateIsSetTo(NMDeviceState.NM_DEVICE_STATE_DISCONNECTED);
+    	thenIsConnectShouldReturn(false);
+    }
+    @Test
+    public void shouldReturnConnectionTruePrepare() throws InterruptedException, KuraException {
+    	whenStateIsSetTo(NMDeviceState.NM_DEVICE_STATE_PREPARE);
+    	thenIsConnectShouldReturn(true);
+    }
+    @Test
+    public void shouldReturnConnectionTrueConfig() throws InterruptedException, KuraException {
+    	whenStateIsSetTo(NMDeviceState.NM_DEVICE_STATE_CONFIG);
+    	thenIsConnectShouldReturn(true);
+    }
+    @Test
+    public void shouldReturnConnectionTrueNeedAuth() throws InterruptedException, KuraException {
+    	whenStateIsSetTo(NMDeviceState.NM_DEVICE_STATE_NEED_AUTH);
+    	thenIsConnectShouldReturn(true);
+    }
+    @Test
+    public void shouldReturnConnectionTrueIpConfig() throws InterruptedException, KuraException {
+    	whenStateIsSetTo(NMDeviceState.NM_DEVICE_STATE_IP_CONFIG);
+    	thenIsConnectShouldReturn(true);
+    }
+    @Test
+    public void shouldReturnConnectionTrueIpCheck() throws InterruptedException, KuraException {
+    	whenStateIsSetTo(NMDeviceState.NM_DEVICE_STATE_IP_CHECK);
+    	thenIsConnectShouldReturn(true);
+    }
+    @Test
+    public void shouldReturnConnectionTrueSecondaries() throws InterruptedException, KuraException {
+    	whenStateIsSetTo(NMDeviceState.NM_DEVICE_STATE_SECONDARIES);
+    	thenIsConnectShouldReturn(true);
+    }
+    @Test
+    public void shouldReturnConnectionTrueActivated() throws InterruptedException, KuraException {
+    	whenStateIsSetTo(NMDeviceState.NM_DEVICE_STATE_ACTIVATED);
+    	thenIsConnectShouldReturn(true);
+    }
+    @Test
+    public void shouldReturnConnectionTrueDeactivating() throws InterruptedException, KuraException {
+    	whenStateIsSetTo(NMDeviceState.NM_DEVICE_STATE_DEACTIVATING);
+    	thenIsConnectShouldReturn(true);
+    }
+    @Test
+    public void shouldReturnConnectionTrueFailed() throws InterruptedException, KuraException {
+    	whenStateIsSetTo(NMDeviceState.NM_DEVICE_STATE_FAILED);
+    	thenIsConnectShouldReturn(true);
+    }
+
+    public void whenInt32StateIsPassed(UInt32 type){
+    	state = NMDeviceState.fromUInt32(type);
+    }
+    
+    public void whenStateIsSetTo(NMDeviceState type){
+    	state = type;
+    }
+    
+    public void thenCorrectStateIsReturned(NMDeviceState type){
+    	assertEquals(state, type);
+    }
+    
+    public void thenIsConnectShouldReturn(Boolean bool){
+    	assertEquals(NMDeviceState.isConnected(state), bool);
+    }
```

Comment:
Just nitpicking here: since you're passing the expected state as a parameter of the function I would name the function as you did for the `thenIsConnectShouldReturn` instead of `thenCorrectStateIsReturned`, which is somewhat less meaningful.

Something like `thenStateShouldReturn` or `thenStateShouldBeEqualTo`...

```suggestion
    public void thenStateShouldBeEqualTo(NMDeviceState type){
    	assertEquals(state, type);
    }
    
    public void thenIsConnectShouldReturn(Boolean bool){
    	assertEquals(NMDeviceState.isConnected(state), bool);
    }
```